mono/mini/exceptions-arm.c

```diff
@@ -14,9 +14,9 @@
 #include <string.h>
 
 #ifndef MONO_CROSS_COMPILE
-#ifdef HAVE_ASM_SIGCONTEXT_H
+#ifdef PLATFORM_ANDROID
 #include <asm/sigcontext.h>
-#endif  /* def HAVE_ASM_SIGCONTEXT_H */
+#endif  /* def PLATFORM_ANDROID */
 #endif
 
```

Comment:
This change looks strange, it would now only include this file if running on Android.

Why was this necessary?

We have two options, if we want it included on Android or if the header file is present, perhaps we should change that to be HAVE_ASM_SIGCONTEXT_H || PLATFORM_ANDROID

If the purpose was to avoid including the header file, I would like to know why :-)